src/python/WMCore/Services/DBS/DBS3Reader.py

```diff
@@ -577,11 +581,12 @@ def listFileBlockLocation(self, fileBlockName, dbsOnly = False, phedexNodes=Fals
                 msg += "%s\n" % str(ex)
                 raise Exception(msg)
 
-            if not blockInfo or len(blockInfo) != len(blockNames): #if we couldnt get data location from PhEDEx, try to look into origin site location from dbs
-                dbsOnly = True
-                blockNames = set(blockNames) - set(blockInfo) #get the blocks we did not find information in phedex
+            if blockInfo and len(blockInfo) == len(blockNames):
+                phedexOnly = True
```

Comment:
The question is not for a single block. The question is when you have a list of blocks in `listFileBlockLocation()`, if it makes sense to return locations from phedex for some blocks and from dbs for other blocks. The current code (with and without my current patch) allows this by default. I.e. it tries to get the locations from phedex first, and if there are some blocks for which it could not get them, it tries to get them from dbs (only for those blocks).